src/routes/auth/index.ts

```diff
@@ -0,0 +1,147 @@
+	} catch (error) {
+		console.error('Error sending email', error);
+		return context.json<StatusResponse>({ message: 'Error sending email' }, StatusCodes.INTERNAL_SERVER_ERROR);
+	}
+});
+
+authRoutes.get('/activate', async (context: Context<EnvironmentBindings>) => {
+	try {
+		const token = context.req.query('token');
+		if (!token) {
+			return context.json({ message: 'Invalid or missing token' }, StatusCodes.BAD_REQUEST);
+		}
+
+		const email = await context.env.ACTIVATION_TOKENS.get(token);
+		if (!email) {
+			return context.json({ message: 'Invalid or expired token' }, StatusCodes.UNAUTHORIZED);
+		}
+
+		const emailExists = await checkEmail(context.env.database, email);
+		if (!emailExists) {
+			return context.json({ message: 'User not found' }, StatusCodes.NOT_FOUND);
+		}
+
+		await activateProfile(context.env.database, email);
+
+		// Remove token after successful activation
+		await context.env.ACTIVATION_TOKENS.delete(token);
+
+		return context.json({ message: 'Account activated successfully' }, StatusCodes.OKAY);
+	} catch (error) {
+		return context.json<StatusResponse>({ message: 'Error activating an account' }, StatusCodes.INTERNAL_SERVER_ERROR);
+	}
+});
+
+authRoutes.post('/sign-in', async (context: Context<EnvironmentBindings>) => {
+	try {
+		const body = await context.req.json();
+		const parsedBody = SignInSchema.parse(body);
+
+		const storedPassword = await getPassword(context.env.database, parsedBody);
+
+		if (!storedPassword) {
+			return context.json<StatusResponse>({ message: 'Unauthorized requests' }, StatusCodes.UNAUTHORIZED);
+		}
+
+		// Extract salt and hashed password from stored format
+		const [saltBase64, storedHashedPassword] = storedPassword.split(':');
+		if (saltBase64 === undefined) {
+			throw new Error('Salt not found');
+		}
+		const salt = new Uint8Array([...atob(saltBase64)].map((c) => c.charCodeAt(0)));
+
+		// Hash input password with the same salt
+		const inputHashedPassword = await hashPasswordPBKDF2(parsedBody.password, salt);
+
+		// Verify if the input password matches the stored password
+		if (inputHashedPassword !== storedHashedPassword) {
+			return context.json<StatusResponse>({ message: 'Unauthorized' }, StatusCodes.UNAUTHORIZED);
+		}
+
+		// Generate session token
+		const sessionToken = crypto.randomUUID();
+		const hoursAhead = 1;
+		const tokenExpiry = String(Date.now() + hoursAhead * 60 * 60 * 1000);
+		const expiryAndUserEmail = `${tokenExpiry} ${parsedBody.email}`;
+		await context.env.SESSION_TOKENS.put(sessionToken, expiryAndUserEmail);
+
+		context.header('Set-Cookie', `auth_token=${sessionToken}; HttpOnly; Secure; SameSite=Strict; Expires=${tokenExpiry}; Path=/; Domain=torontojs.com`);
+
+		return context.json<StatusResponse>({ message: 'Authorized successfully', data: sessionToken }, StatusCodes.CREATED);
+	} catch (err) {
+		if (err instanceof ZodError) {
+			return context.json<StatusResponse>({ message: err?.message ?? 'Invalid input' }, StatusCodes.BAD_REQUEST);
```

Comment:
422 > 400

Reply:
422 represents validation errors closer than 400?

Reply:
Yes, 422 is for: "I could parse the body (as JSON) but the values sent to didn't match". 415 is if it's the wrong media type (not JSON). `400` is good if for example JSON itself was invalid, but it's also the error code if you can't find a better, more specific 4xx code that suits.

Reply:
Got it. That makes sense